File: Greedy/javascript/11047.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
let [n, k] = input[0].split(" ").map(Number);
const coins = input
  .slice(1, input.length - 1)
  .map(Number)
  .sort((a, b) => b - a);
let cnt = 0;
for (coin of coins) {
  if (k === 0) break;
  if (coin > k) continue;
  cnt += Math.floor(k / coin);
  k %= coin;
}

console.log(cnt);

File: Greedy/python/11047.py
n,k = map(int,input().split())
coins = [int(input()) for _ in range(n)]
cnt = 0
for i in range(len(coins)-1,-1,-1):
  if k==0 : break
  if coins[i] >k : continue
  cnt += k//coins[i]
  k %= coins[i]
print(cnt)